ts/features/idpay/barcode/store/index.ts

```diff
@@ -0,0 +1,64 @@
+import { pot } from "@pagopa/ts-commons";
+import { createSelector } from "reselect";
+import { getType } from "typesafe-actions";
+import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
+import { TransactionErrorDTO } from "../../../../../definitions/idpay/TransactionErrorDTO";
+import { Action } from "../../../../store/actions/types";
+import { GlobalState } from "../../../../store/reducers/types";
+import { NetworkError } from "../../../../utils/errors";
+import { idPayGenerateBarcode } from "./actions";
+
+export type IdPayBarcodeState = {
+  [initiativeId: string]: pot.Pot<
+    Omit<TransactionBarCodeResponse, "initiativeId">,
+    TransactionErrorDTO | NetworkError
+  >;
+};
```

Comment:
I think we should store the complete object with the `initiativeId`.

```suggestion
export type IdPayBarcodeState = {
  [initiativeId: string]: pot.Pot<TransactionBarCodeResponse, TransactionErrorDTO | NetworkError>;
};
```

Reply:
while theoretically it does make more sense to not have overhead in modifying the response, wouldn't it be redundant for no reason?
After all, to access the store you need an `initiativeId`, and you'd never need that field at all.

Reply:
Imagine if we ever should pass the object to some other screen/saga, in that case we have an incomplete object and we probably should gather the missing information somewhere else. This does not make much sense to me.
Redundancy is never a problem, imho.

Reply:
welp yes, at the end of the day it's just one more field.
addressed in [b5a3ebd](https://github.com/pagopa/io-app/pull/5212/commits/b5a3ebd2e0a64892024da455f4ffb2510d118b49) 👍